src/containers/track-page/store/sagas.js

```diff
@@ -8,52 +8,35 @@ import { tracksActions } from './lineups/tracks/actions'
 import { waitForBackendSetup } from 'store/backend/sagas'
 import { getIsReachable } from 'store/reachability/selectors'
 import { getTrack as getCachedTrack } from 'store/cache/tracks/selectors'
-import { getTrack, getUser } from './selectors'
-import TimeRange from 'models/TimeRange'
+import { getTrack, getTrendingTrackRanks, getUser } from './selectors'
 import { push as pushRoute } from 'connected-react-router'
 import { retrieveTracks } from 'store/cache/tracks/utils'
 import { NOT_FOUND_PAGE, trackRemixesPage } from 'utils/route'
 import { getUsers } from 'store/cache/users/selectors'
-import { retrieveTrending } from 'containers/track-page/store/retrieveTrending'
+import apiClient from 'services/audius-api-client/AudiusAPIClient'
 
 export const TRENDING_BADGE_LIMIT = 10
 
 function* watchTrackBadge() {
   yield takeEvery(trackPageActions.GET_TRACK_RANKS, function* (action) {
     yield call(waitForBackendSetup)
-    const [
-      weeklyTrendingTracks,
-      monthlyTrendingTracks,
-      yearlyTrendingTracks
-    ] = yield all([
-      call(retrieveTrending, {
-        timeRange: TimeRange.WEEK,
-        offset: 0,
-        limit: TRENDING_BADGE_LIMIT,
-        genre: null
-      }),
-      call(retrieveTrending, {
-        timeRange: TimeRange.MONTH,
-        offset: 0,
-        limit: TRENDING_BADGE_LIMIT,
-        genre: null
-      }),
-      call(retrieveTrending, {
-        timeRange: TimeRange.YEAR,
-        offset: 0,
-        limit: TRENDING_BADGE_LIMIT,
-        genre: null
+    let trendingTrackRanks = yield select(getTrendingTrackRanks)
+    if (trendingTrackRanks.week === null) {
```

Comment:
Nit, prefer this logic to be earlier up, maybe just in the selector. You could do

if !(ttR.week && ttR.month && ttR.year) { return null } and then in this check here you could just check trendingTrackRanks !== null or w/e